test/unit/manipulation.js

```diff
@@ -500,6 +500,53 @@ QUnit.test( "html(String) tag-hyphenated elements (Bug #1987)", function( assert
 	assert.equal( j.children().text(), "text", "Tags with multiple hypens behave normally" );
 } );
 
+QUnit.test( "html(String) respect the spec for tag names (gh-2005) ", function( assert ) {
```

Comment:
Thank you for this! I think my only note is to be more specific about how we're respecting the spec here.